src/app/utils/reorderable-list/reorderable-list.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, EventEmitter, Input, Output } from "@angular/core";
+import {
+  CdkDropList,
+  CdkDrag,
+  CdkDragDrop,
+  moveItemInArray,
+} from "@angular/cdk/drag-drop";
+import { CommonModule } from "@angular/common";
+import { ColumnConfig } from "app/core/common-components/entity-form/FormConfig";
+import { EntityConstructor } from "#src/app/core/entity/model/entity";
+import { EntityFieldsMenuComponent } from "#src/app/core/common-components/entity-fields-menu/entity-fields-menu.component";
+import { EntityFieldLabelComponent } from "#src/app/core/common-components/entity-field-label/entity-field-label.component";
+import { FontAwesomeModule } from "@fortawesome/angular-fontawesome";
+import { MatFormFieldModule } from "@angular/material/form-field";
+import { MatSelectModule } from "@angular/material/select";
+
+@Component({
+  selector: "app-reorderable-list",
+  imports: [
+    CommonModule,
+    CdkDropList,
+    CdkDrag,
+    EntityFieldsMenuComponent,
+    EntityFieldLabelComponent,
+    FontAwesomeModule,
+    MatFormFieldModule,
+    MatSelectModule,
+  ],
+  templateUrl: "./reorderable-list.component.html",
+  styleUrl: "./reorderable-list.component.scss",
+})
+export class ReorderableListComponent {
+  @Input() items: any[] = [];
+  @Input() availableItems: (string | ColumnConfig)[] = [];
+  @Input() entityType: EntityConstructor;
+  @Input() orientation: "vertical" | "horizontal" | "mixed" = "vertical";
+  @Input() dropLockAxis: "x" | "y" | null = null;
```

Comment:
looks like currently we are always using this with "mixed" orientation. Do we already see a use case where this will need to be changed?

Otherwise, let's remove them from the "public interface" (i.e. not make them `@Input()` to keep things simple for now.